jetty-ee10/jetty-ee10-servlet/src/test/java/org/eclipse/jetty/ee10/servlet/DefaultServletCombinationsTest.java

```diff
@@ -0,0 +1,258 @@
+
+        Path subdirPath = docRoot.resolve("subdir");
+        FS.ensureDirExists(subdirPath);
+        Files.writeString(subdirPath.resolve("index.html"), "Static index.html at root subdir", UTF_8);
+        Files.writeString(subdirPath.resolve("foo.welcome"), "Static foo.welcome at root subdir", UTF_8);
+
+        Path emptyPath = docRoot.resolve("empty");
+        FS.ensureDirExists(emptyPath);
+
+        Path staticPath = docRoot.resolve("static");
+        FS.ensureDirExists(staticPath);
+        Files.writeString(staticPath.resolve("index.html"), "Static index.html at static", UTF_8);
+        Files.writeString(staticPath.resolve("foo.welcome"), "Static foo.welcome at static", UTF_8);
+
+        Path staticsubdirPath = staticPath.resolve("subdir");
+        FS.ensureDirExists(staticsubdirPath);
+        Files.writeString(staticsubdirPath.resolve("index.html"), "Static index.html at static subdir", UTF_8);
+        Files.writeString(staticsubdirPath.resolve("foo.welcome"), "Static foo.welcome at static subdir", UTF_8);
+
+        Path subdirHtmlPath = docRoot.resolve("subdirHtml");
+        FS.ensureDirExists(subdirHtmlPath);
+        Files.writeString(subdirHtmlPath.resolve("index.html"), "Static index.html at root subdirHtml", UTF_8);
+        Path staticSubdirHtmlPath = staticPath.resolve("subdirHtml");
+        FS.ensureDirExists(staticSubdirHtmlPath);
+        Files.writeString(staticSubdirHtmlPath.resolve("index.html"), "Static index.html at static subdirHtml", UTF_8);
+
+        Path subdirWelcomePath = docRoot.resolve("subdirWelcome");
+        FS.ensureDirExists(subdirWelcomePath);
+        Files.writeString(subdirWelcomePath.resolve("foo.welcome"), "Static foo.welcome at root subdirWelcome", UTF_8);
+
+        Path staticSubdirWelcomePath = staticPath.resolve("subdirWelcome");
+        FS.ensureDirExists(staticSubdirWelcomePath);
+        Files.writeString(staticSubdirWelcomePath.resolve("foo.welcome"), "Static foo.welcome at static subdirWelcome", UTF_8);
+
+        Path subdirEmptyPath = staticPath.resolve("subdirEmpty");
+        FS.ensureDirExists(subdirEmptyPath);
+
+        server = new Server();
+
+        connector = new LocalConnector(server);
+        connector.getConnectionFactory(HttpConfiguration.ConnectionFactory.class).getHttpConfiguration().setSendServerVersion(false);
+
+        context = new ServletContextHandler();
+        context.setBaseResourceAsPath(docRoot);
+        context.setContextPath("/");
+        context.setWelcomeFiles(new String[]{"index.html", "index.welcome"});
+
+        ServletHolder welcomeExtHolder = context.addServlet(WelcomeServlet.class, "*.welcome");
+        welcomeExtHolder.setInitParameter("mapping", "welcome extension");
+    }
+
+    private void startServer(boolean pathInfoOnly, ResourceService.WelcomeMode welcomeMode) throws Exception
+    {
+        ServletHolder defaultHolder = context.addServlet(DefaultServlet.class, "/static/*");
```

Comment:
I think this should be:
```suggestion
        ServletHolder defaultHolder = context.addServlet(DefaultServlet.class, pathInfoOnly ? "/static/*" : "/");
```

Reply:
Done.

Reply:
Ah, I see, you actually have 2 DefaultServlet instances, one that really is a default servlet always at "/" and another that is configured at "/static/*"

I think the test should be changed to only ever have 1 `DefaultServlet`, which is at "/" when `pathInfoOnly` is false. otherwise it is at "/static/*" and some error servlet should be at "/".

Then there needs to be no tests that explicitly have "/static" in their URI. Instead, if `pathInfoOnly` is true, you should prefix "/static" to all the tests and check all the combinations work if they are not at the root and have pathInfo only.


Hmmm but then we might need another combination.  Testing a non default DefaultServlet, but that has `pathInfoOnly` false!     Ugh sooo many variations!  But that is the point of this unit test.

Reply:
Yes, your suggestion would make the combinations more readable. Let me try this...